tests/libsample/objecttype.h

```diff
@@ -70,7 +70,15 @@ class LIBSAMPLE_API ObjectType
     Str objectName() const;
     void setObjectName(const Str& name);
 
-    inline unsigned long identifier() const { return reinterpret_cast<unsigned long>(this); }
+    inline unsigned long identifier() const {
+#if __SIZEOF_POINTER__ == 8
```

Comment:
this one is identifier. Not the hash. So it's bad idea to drop bytes here. It's only two options here

- change return type checking all the usages
- add the static map index `unsigned long` to `uintptr_t` and back.

For me the first one looks better